VimeoUpload/Upload/Descriptor System/Cameo/CAMUploadDescriptor.swift

```diff
@@ -0,0 +1,356 @@
+    {
+        let sessionManager = sessionManager as! VimeoSessionManager
+        let responseSerializer = sessionManager.responseSerializer as! VimeoResponseSerializer
+        
+        do
+        {
+            switch self.currentRequest
+            {
+            case .CreateVideo:
+                self.uploadTicket = try responseSerializer.processCreateVideoResponse(task.response, url: url, error: error)
+                
+            case .UploadVideo:
+                break
+                
+            case .ActivateVideo:
+                self.videoUri = try responseSerializer.processActivateVideoResponse(task.response, url: url, error: error)
+                
+            case .VideoSettings:
+                self.video = try responseSerializer.processVideoSettingsResponse(task.response, url: url, error: error)
+                
+            case .CreateThumbnail:
+                self.thumbnailTicket = try responseSerializer.processCreateThumbnailResponse(task.response, url: url, error: error)
+                
+            case .UploadThumbnail:
+                break
+                
+            case .ActivateThumbnail:
+                self.thumbnail = try responseSerializer.processActivateThumbnailResponse(task.response, url: url, error: error)
+            }
+        }
+        catch let error as NSError
+        {
+            self.error = error
+            self.currentTaskIdentifier = nil
+            self.state = .Finished
+        }
+        
+        return nil
+    }
+    
+    override func taskDidComplete(sessionManager sessionManager: AFURLSessionManager, task: NSURLSessionTask, error: NSError?)
+    {
+        if self.currentRequest == .UploadVideo
+        {
+            NSFileManager.defaultManager().deleteFileAtURL(self.videoUrl)
+        }
+        else if self.currentRequest == .UploadThumbnail
+        {
+            if let thumbnailFileUrl = self.thumbnailFileUrl {
+                NSFileManager.defaultManager().deleteFileAtURL(thumbnailFileUrl)
+            }
+        }
+        
+        if self.error == nil
+        {
+            if let taskError = task.error // task.error is reserved for client-side errors, so check it first
+            {
+                let domain = self.errorDomainForRequest(self.currentRequest)
+                self.error = taskError.errorByAddingDomain(domain)
+            }
+            else if let error = error
+            {
+                let domain = self.errorDomainForRequest(self.currentRequest)
+                self.error = error.errorByAddingDomain(domain)
+            }
+        }
+        
+        var nextRequest = CAMUploadRequest.nextRequest(self.currentRequest)
+        if self.error != nil || nextRequest == nil
+        {
+            self.currentTaskIdentifier = nil
+            self.state = .Finished
+            
+            return
+        }
+        else if nextRequest == .VideoSettings && self.videoSettings == nil
```

Comment:
Is there a way to set up these `else if`s and the `nextRequest` var so (1) this passage is a little easier to read/work with and (2) we don't have to force unwrap `nextRequest` in a few places?